manifests/init.pp

```diff
@@ -140,6 +144,7 @@
     config_hash => $config_hash_real,
     purge       => $purge_config_dir,
     notify      => $notify_service,
+    rpc_port    => $custom_rpc_port,
```

Comment:
I don't think you need to pass this as a param, you can just do:

```
if ($config_hash_real['ports'] and $config_hash_real['ports']['rpc']) {
  $rpc_port = $config_hash_real['ports']['rpc']
} else {
  $rpc_port = 8400
}
```

then while calling it in `config.pp` just do `$consul::rpc_port`